opencensus/OpenCensus-Traces.md

```diff
@@ -0,0 +1,27 @@
+# OpenCensus Trace Exporters
+
+## Exporting Spans
+
+Each span received by the OpenCensus Exporter must be transformed into a New Relic span before being sent.
+
+|NR Span Field|OpenCensus Span Field|Type|Required|Comments|
+|-----|-----|-----|-----|-----|
+|`id`|`Span.SpanID`|string|yes|Unique identifier for this span|
+|`trace.id`|`Span.SpanContext.TraceID`|string|yes|Unique identifier shared by all spans within a single trace|
+|`name`|`Span.Name`|string|yes|The name of this span|
+|`parent.id`|`Span.ParentSpanID`|string|yes (except root span)|The span id of the previous caller of this span. Can be empty if this is the first span in which case the key should be omitted.|
+|`timestamp`|`Span.StartTime`|long|yes|Epoch ms timestamp|
+|`duration.ms`|`Span.EndTime` - `Span.StartTime`|float|yes|Duration of this span in milliseconds|
+|`service.name`|(none)|string|yes| The name of the service that created this span. This value is not gathered from OpenCensus. Instead it should be set directly by the user when defining the Exporter.|
+|`attributes`|`Span.Attributes`|map|no|Map of user specified "tags" on this span. Keys are strings, values can be any of bool, long, float, or string. Key should be omitted if empty.|
+|`attributes.error`|`Span.Status.Code` not in `IgnoreStatusCodes` list and is not 0|bool|no|When `Span.Stats.Code` is not `0` and is not in the `IgnoreStatusCodes` list, include the value `"error": true` in the `tags` map. If an `error` tag was already set by the user, do not override it.|
```

Comment:
do we have access to an error message? If so, we should also populate `error.message` here.

Reply:
Oh great idea! We didn't do this in our version of the Go OpenCensus exporter but it's a great idea. I'll add it!

Since the [`Status`](https://godoc.org/go.opencensus.io/trace#Status) isn't always an error (say the code is 0) then maybe we shouldn't use the word "error". How about three attributes (all under the attributes map) `error` (bool), `status.code` (int), and `status.message` (string)?